src/tools/rustfmt/src/bin/main.rs

```diff
@@ -202,7 +202,11 @@ fn is_nightly() -> bool {
 
 // Returned i32 is an exit code
 fn execute(opts: &Options) -> Result<i32> {
-    let matches = opts.parse(env::args().skip(1))?;
+    let matches = match rustc_driver::args::raw_args() {
+        Ok(args) => opts.parse(args.into_iter().skip(1))?,
+        // Error message has already been printed, so just return an error code
+        Err(_) => return Ok(rustc_driver::EXIT_FAILURE),
+    };
```

Comment:
This seems wrong to me in rustfmt's particular context. clippy and miri are both actually using rustc_driver whereas rustfmt isn't (there's only been a _very_ recent change to utilize it just for the ice bug report url).

If non-unicode args are causing an issue in rustfmt as well, then I feel like that's something that should be fixed upstream in `getopts` instead of worked around here by introducing more rustc_driver usage.

I might be open to considering this as a tactical approach while getopts is being addressed, but only if there's some type of real/pressing issue with this behavior in rustfmt specifically

Reply:
Thanks @beetrees, but I still hold the position I articulated in my prior comment and would like to see the rustfmt changes removed from this PR.

To be clear, I agree this is something that should get resolved, and appreciate you bringing it to attention. However, this path with these change here in-tree isn't the route I'd want to take (barring any other information)